### ISSUE TITLE
Error message

### ISSUE THREAD

**Issue by jsreaves** [gregorias/anki-code-highlighter#34]
Hello,

I'm getting the error message below when I highlight a code snippet and then click the icon. When I click the icon without anything highlighted, I get the expected dialog boxes to choose the highlighter and language.

Debug info:
Anki 2.1.56 (07fd88dd) Python 3.9.15 Qt 6.3.2 PyQt 6.3.1
Platform: macOS-13.1-arm64-arm-64bit
Flags: frz=True ao=True sv=3
Add-ons, last update check: 2023-01-21 06:48:20
Add-ons possibly involved: ⁨Highlight Code⁩

Caught exception:
Traceback (most recent call last):
  File "aqt.webview", line 562, in handler
  File "/Users/username/Library/Application Support/Anki2/addons21/112228974/ankieditorextra.py", line 76, in transform_field
    field = extract_field_from_web_editor(web_editor_html)
  File "/Users/username/Library/Application Support/Anki2/addons21/112228974/ankieditorextra.py", line 39, in extract_field_from_web_editor
    result = re.search('<anki-editable[^>]*>(.*)</anki-editable>',
  File "re", line 201, in search
TypeError: expected string or bytes-like object

**Comment by gregorias** [comment]
Hi, could you provide more detailed steps to reproduce this? To mitigate the impact of this bug, could you try the keyboard shortcut (Command + ')?

**Comment by jsreaves** [comment]
1. Paste code into Front field of Basic Anki card template using shift-command-V to keep formatting. 
2. Click the icon for the plug in.
3. Error dialog pops up.

The same thing happens when I use the keyboard shortcut (Cmd + ') instead of the icon. The reason I was using the icon in the first place is that I mis-read the keyboard shortcut and was trying (Cmd + +), which seems to change the font size.

I also tried disabling all other add-ons, but unfortunately, this didn't fix the problem.

As you can see from the error message,I am using Anki 2.1.56, which just came out.

**Comment by jsreaves** [comment]
I am able to reproduce the error when I paste in an example from this page:  https://www.codecademy.com/resources/docs/python/recursion

I think the error depends on how the code snippet is formatted. For instance, when I paste code the example from the Python wiki page (with the shift key, to keep formatting), I do not get the error message I reported yesterday, but I do get a breakdown of the HTML tags when I click the code highlighting icon. Below is what appears in the Front field of Anki Add screen after I click the highlighting plugin.  (This snippet shows what appears in normal mode, not in HTML editor mode.) As you can see, HTML is breaking through.

Maybe retained HTML tags in pasted code snippets are causing the ```re.search``` to fail, or to otherwise misbehave? An obvious solution would be to strip formatting when pasting, but this would force the user to add back line breaks and indentation.

```
n = int(input('Type a number, and its factorial will be printed: '))

<span style="font-weight: bold;">if</span> n &lt; 0:
    <span style="font-weight: bold;">raise</span> <span style="font-weight: bold;">ValueError</span>('You must enter a non-negative integer')

factorial = 1
<span style="font-weight: bold;">for</span> i <span style="font-weight: bold;">in</span> range(2, n + 1):
    factorial *= i

print(factorial)
```

**Comment by gregorias** [comment]
Thank you, but could you provide more details? I've tried 2.1.56 and couldn't reproduce the bug. What I need is:
1. The HTML code of the field in which you make a selection.
2. A screenshot of the selection before running this plugin.

Also note that [this plugin only works on untagged code](https://github.com/gregorias/anki-code-highlighter#:~:text=the%20plugin%20works%20on%20untagged%20code%20blocks.%20do%20not%20add%20your%20own%20html%20tags%20like%20%3Ccode%3E%20or%20%3Cpre%3E%3Ccode%3E%20to%20the%20snippet%20you%20want%20to%20highlight.%20if%20you%20have%20run%20into%20issues%20with%20preannotated%20code%20snippets%2C%20see%20this%20comment%20for%20how%20to%20fix%20this.). You need to clean pasted code from HTML tags, because this plugin works on the HTML source and not on the rendered text you see in the WYSIWYG editor.

**Comment by jsreaves** [comment]
In my Anki preferences, I have enabled "Paste without shift key strips formatting." My initial assumption was that pasting in this way would remove HTML tags and allow the plugin to work. However, I have found that some tags are _not_ stripped, and this leads to the error. In my tests, pasting without the shift key strips the ```<pre>``` tag but does not strip ```<div>``` tags. The error occurs when either of these tags is present.

<img width="636" alt="Screenshot 2023-01-24 at 7 35 26 AM" src="https://user-images.githubusercontent.com/12678756/214293464-7c96fe90-d2e2-42dc-a70a-c4dc7fe46348.png">

Example 1: Paste with Cmd-V (strip formatting)  -> Error
<img width="1374" alt="Screenshot 2023-01-24 at 7 21 12 AM" src="https://user-images.githubusercontent.com/12678756/214290759-d7eef7be-861f-4b8f-bba3-36e9c4f2f22e.png">

Example 2: Paste with Cmd-Shift-V -> Error
<img width="1376" alt="Screenshot 2023-01-24 at 7 26 41 AM" src="https://user-images.githubusercontent.com/12678756/214291580-da762ec8-2bb1-4890-8bb2-7cc303644454.png">

Example 3: Paste to text editor, then paste to Anki -> No error
<img width="1377" alt="Screenshot 2023-01-24 at 7 29 01 AM" src="https://user-images.githubusercontent.com/12678756/214291983-dfbb971b-bfa2-4685-868c-0652e0f7db5c.png">
